File: backend/app/models/private_equity.py
from app.models.common import CamelModel


class Investissement(CamelModel):
    nom: str
    vintage: int
    investi: int
    valorisation: int
    tvpi: float
    statut: str


class TimelineEvent(CamelModel):
    date: str
    type: str
    montant: int
    fonds: str


class PrivateEquityData(CamelModel):
    investi: int
    valorisation: int
    tvpi: float
    dpi: float
    irr_estime: float
    fonds: list[Investissement]
    timeline: list[TimelineEvent]

File: frontend/src/types/private-equity.ts
export interface Investissement {
  nom: string;
  vintage: number;
  investi: number;
  valorisation: number;
  tvpi: number;
  statut: string;
}

export interface TimelineEvent {
  date: string;
  type: string;
  montant: number;
  fonds: string;
}

export interface PrivateEquityData {
  investi: number;
  valorisation: number;
  tvpi: number;
  dpi: number;
  irrEstime: number;
  fonds: Investissement[];
  timeline: TimelineEvent[];
}
